contracts/cw2981-royalties/src/lib.rs

```diff
@@ -80,7 +80,40 @@ pub mod entry {
         info: MessageInfo,
         msg: ExecuteMsg,
     ) -> Result<Response, ContractError> {
-        Cw2981Contract::default().execute(deps, env, info, msg)
+        match msg {
+            ExecuteMsg::Mint {
+                token_id,
+                owner,
+                token_uri,
+                extension,
+            } => {
+                // validate royalty_percentage to be between 0 and 100
+                if let Some(Metadata {
+                    royalty_percentage: Some(royalty_percentage),
+                    ..
+                }) = extension
+                {
+                    // no need to check < 0 because royalty_percentage is u64
+                    if royalty_percentage > 100 {
+                        return Err(ContractError::Std(StdError::GenericErr {
```

Comment:
We should define our own error type instead of inheriting from cw721-base. Let me see if I can push a commit

Reply:
@peara I pushed a commit, please check: https://github.com/CosmWasm/cw-nfts/pull/111/commits/db45da19e387d779992f076101e6137d85c1d8bd

Reply:
Thank you, it looks much cleaner.